src/Graphs/DirectedAcyclicGraphTransitiveReducer.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Bearded.Utilities.Graphs
+{
+    public static class DirectedAcyclicGraphTransitiveReducer<T> where T : IEquatable<T>
+    {
+        public static IDirectedAcyclicGraph<T> ReduceGraph(IDirectedAcyclicGraph<T> graph)
+        {
+            var elements = graph.Elements as IList<T> ?? graph.Elements.ToList();
+            var graphBuilder = DirectedGraphBuilder<T>.NewBuilder();
+
+            foreach (var element in elements)
+            {
+                graphBuilder.AddElement(element);
+            }
+
+            foreach (var element in elements)
+            {
+                foreach (var neighbor in getOnlyDirectChildren(graph, element))
+                {
+                    graphBuilder.AddArrow(element, neighbor);
+                }
+            }
+
+            return graphBuilder.CreateAcyclicGraphUnsafe();
+        }
+
+        private static IEnumerable<T> getOnlyDirectChildren(IDirectedGraph<T> graph, T element)
+        {
+            var children = graph.GetDirectSuccessorsOf(element) as IList<T>
+                ?? graph.GetDirectSuccessorsOf(element).ToList();
```

Comment:
I'd prefer caching the result of this method I think. Though we do much worse than this in the recursive function below I suppose...